tests/actions/ReportPreviewActionUtilsTest.ts

```diff
@@ -0,0 +1,155 @@
+import type {OnyxCollection} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import getReportPreviewAction from '@libs/ReportPreviewActionUtils';
+// eslint-disable-next-line no-restricted-syntax
+import * as ReportUtils from '@libs/ReportUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Policy, Report, ReportViolations, Transaction, TransactionViolation} from '@src/types/onyx';
+
+const CURRENT_USER_ACCOUNT_ID = 1;
+const CURRENT_USER_EMAIL = 'tester@mail.com';
+
+const SESSION = {
+    email: CURRENT_USER_EMAIL,
+    accountID: CURRENT_USER_ACCOUNT_ID,
+};
+
+const PERSONAL_DETAILS = {
+    accountID: CURRENT_USER_ACCOUNT_ID,
+    login: CURRENT_USER_EMAIL,
+};
+
+const REPORT_ID = 1;
+const TRANSACTION_ID = 'TRANSACTION_ID';
+const REPORT_TRANSACTIONS = {} as Transaction[];
+const VIOLATIONS = {} as OnyxCollection<TransactionViolation[]>;
```

Comment:
We shouldn't cast the data type from an object to an array.

```suggestion
const REPORT_TRANSACTIONS: Transaction[] = [];
const VIOLATIONS: OnyxCollection<TransactionViolation[]> = {};
```